pkg/server/server.go

```diff
@@ -378,3 +378,41 @@ func (s *Server) RuntimeHook(ctx context.Context, req *tetragon.RuntimeHookReque
 	}
 	return &tetragon.RuntimeHookResponse{}, nil
 }
+
+func (s *Server) GetLogLevel(ctx context.Context, req *tetragon.GetLogLevelRequest) (*tetragon.GetLogLevelResponse, error) {
+	logger.GetLogger().Debugf("Client requested current log level: %s", logger.GetLogLevel().String())
+	return &tetragon.GetLogLevelResponse{Level: logger.GetLogLevel().String()}, nil
+}
+
+func (s *Server) SetLogLevel(ctx context.Context, req *tetragon.SetLogLevelRequest) (*tetragon.SetLogLevelResponse, error) {
+	currentLogLevel := logger.GetLogLevel()
+	changedLogLevel, err := logrus.ParseLevel(req.GetLevel())
+	if err != nil {
+		return nil, fmt.Errorf("invalid log level: %s", req.GetLevel())
+	}
+
+	// Temporarily set the logger to the lowest level to ensure the message is logged
+	logger.SetLogLevel(logrus.TraceLevel)
```

Comment:
Will this impact other routines also using logger here? I'm not sure about internals of logging but we wouldn't want to shortly spew debug/trace info if another sensor is running.

Why not just make it a Warn if its critical to have printed? Seems cleaner than bouncing loglevel around.

Reply:
Thanks for input. Addressed via https://github.com/cilium/tetragon/pull/2643/commits/d0c4133f016a1f4533a898440a4d22ad8d031f57.